Meson build system

telegraphic opened the pull request:
Opening PR to use for notes and todos

telegraphic committed:
Adding meson build take 1

telegraphic commented:
TODO: Read `config.h.in` and generate `config.h`

eli-schwartz commented on meson.build:
```diff
@@ -0,0 +1,59 @@
+
+# Basic setup -- We should be able to just add
+#    default_options: ['debug=true', 'cpp_std=c++14']
+# to the project() call, but nvcc does not appear to get these passed on
```
I guess this would work with `cuda_std=c++14` though?

Commit:
Adding files from blpc0

Commit:
Adding stringify as subproject

telegraphic committed:
Added config.h generation, still working on stringify

telegraphic committed:
Updates to meson.build (tidyup)

Adding meson.build and stringify

telegraphic committed:
First version to compile shared object!

telegraphic committed:
stringify wrap now works, adding custom_target for cufft_static

telegraphic committed:
Now building using cufft static dlink

See https://github.com/mesonbuild/meson/discussions/11411

telegraphic committed:
configure template: stringify.sh to stringify.sh.in

Adding bifrost/ prefix to config.h; run stringify

Got stringifier subproject working with find_program

Typo fix

Typo fix

Moving call to stringify

Changes for stringify execution order

Yet another attempt to stringify in meson

More sane approach to stringify JIT compilation

Updates to stringify more sane approach

Updates to meson.build adding meson_options

telegraphic committed:
Adding python installation capability

telegraphic committed:
Adding TODOs

telegraphic committed:
Moving python to its own meson.build

telegraphic committed:
Moved headers to include/

telegraphic committed:
Rejig of subdirs and includes

telegraphic committed:
Adding summary to meson

telegraphic commented:
TODO: need to copy libbifrost_generated.py into python/libbifrost_generated.py

telegraphic commented:
TODO: Make CUDA flags automatic

telegraphic commented:
TODO: install_headers into subdir bifrost
